core/src/consensus.rs

```diff
@@ -400,9 +462,20 @@ impl Tower {
         total_stake: u64,
         epoch_vote_accounts: &HashMap<Pubkey, (u64, Account)>,
     ) -> SwitchForkDecision {
+        let mut maybe_stray_ancestors = HashSet::new();
+
         self.last_voted_slot()
             .map(|last_voted_slot| {
-                let last_vote_ancestors = ancestors.get(&last_voted_slot).unwrap();
+                let last_vote_ancestors = ancestors.get(&last_voted_slot).unwrap_or_else(|| {
+                    if self.is_restored_stray_slot(last_voted_slot) {
+                        // Construct artifical ancestors because we can't with ancestors from bank_forks
+                        maybe_stray_ancestors = self.restored_stray_slots.range(0..last_voted_slot).copied().collect();
```

Comment:
write a test for this, showing different behavior with previous implementation (returning HashSet::default()).

Reply:
done!